### ISSUE TITLE
Type narrowing using "in tuple"

### ISSUE THREAD

**Issue by yabirgb** [python/mypy#12123]
Mypy fails to recognize after a explicit check that a set of variables can't be None

**To Reproduce**

Run mypy on the following script

```python
from typing import Optional

class Test:    
    def double(self, x: int) -> int:
        return x * 2

def check_value(a: Optional[Test], b: Optional[int]) -> Optional[int]:
    if None in (a, b):
        return None
    
    return a.double(b)
    
if __name__ == '__main__':
    check_value(Test(), 2)
```

**Expected Behavior**

The linter should not complain as after the if check both variables `a` and `b` can't be `None`.

**Actual Behavior**

The linter says that the `None` possibility for the variable lacks the method. Is not correct since we have
taken action to prevent this.

```
testmypy.py:11: error: Item "None" of "Optional[Test]" has no attribute "double"
testmypy.py:11: error: Argument 1 to "double" of "Test" has incompatible type "Optional[int]"; expected "int"
```

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags: `mypy testmypy.py`
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.9.10 and 3.7
- Operating system and version: `EndeavourOS Linux x86_64 - 5.16.4-arch1-1`

**Comment by erictraut** [comment]
The pattern `None in (a, b)` is not current supported for type narrowing by mypy. If you use the pattern `a is None or b is None`, it will work as expected.

**Comment by nunesvictor** [comment]
I have a similar problem. See, if I try to check if several elements are not None with all function, mypy ins't capable to infer the variable isn't Optional anymore. Example:

```python
my_model: Optional[MyModel] = optionalDict.get('my_model')
my_second_model Optional[MySecondModel] = optionalDict.get('my_second_model')

if not all([my_model, my_second_model]):
    raise BaseException('Models cannot be None')

my_model.attr # <--- still infers like Optional[MyModel]
```

**Comment by JelleZijlstra** [comment]
Duplicate of #3229
